BudgetSquirrel.Api/Infrastructure/UnitOfWork.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Threading.Tasks;
+using BudgetSquirrel.Business;
+using BudgetSquirrel.Business.BudgetPlanning;
+using BudgetSquirrel.Data.EntityFramework;
+
+namespace BudgetSquirrel.Api.Infrastructure
+{
+  public class UnitOfWork : IUnitOfWork
```

Comment:
So this is specific to budget, do we need to create a separate one for transactions?

Reply:
This isn't specific to anything. It just basically abstracts the write operations of the Entity Framework DBContext and provides a way to access the DbSets on the context (using GetRepository).

Reply:
Looks like you are required to pass in a `IRepository<Budget>>` Perhaps this should be `UnitOfWork<T>` when you can define the repository model. Also we should update `budgetRepository` to be just `repository`